### DISCUSSION TITLE
Handling Exceptions

### DISCUSSION THREAD

**Discussion by BasselAshi** [Auties00/WhatsappWeb4j#409] [Q&A]
Since the app runs on a different thread, any way we can handle the runtime exceptions being thrown?

**Comment by Auties00** [comment]
> Since the app runs on a different thread, any way we can handle the runtime exceptions being thrown?

Implement an error handler and pass it to the api, for async operations use exceptionally

**Reply by BasselAshi** [reply]
This will only work if the exception I am trying to catch is during CompletableFuture<Whatsapp>. However, the errors I am getting are happening later when nodes are being sent/received or when the socket suddently closes.

**Reply by Auties00** [reply]
> This will only work if the exception I am trying to catch is during CompletableFuture<Whatsapp>. However, the errors I am getting are happening later when nodes are being sent/received or when the socket suddently closes.

As I said you need to implement an error handler and pass it to the api
